### DISCUSSION TITLE
1.4.3 / 1.4.11 Meeting contrast requirements via conforming alternate versions?

### DISCUSSION THREAD

**Discussion by detlevhfischer** [w3c/wcag#3827] [Q&A]
The question is (or seems) simple.
Websites with text / graphics below the contrast requirements of 1.4.3 and 1.4.11 often offer contrast switches to load different styles that, after activation, increase contrast.

I assume that the option to meet contrast requirements via activating a fully accessible contrast switch is considered by many as sufficient to meet 1.4.3 / 1.4.11 (provided the mechanism is fully accessible and the resulting contrast of content rendered in the alternate version meets the requirements). 

**BUT:** 5.2.1 Conformance level requires that one of the (...) levels of conformance is **met in full**, which means for AA:
> For Level AA conformance, the Web page satisfies all the Level A and Level AA Success Criteria, or a Level AA conforming alternate version is provided."

This will often not be the case for a contrast switch that makes contrast conform but does not remedy other existing level AA issues / FAILs of the page. Would this mean that a contrast switch can only be assumed to meet 1.4.3 / 1.4.11 if no other issues / FAILs remain (i.e., the site in its default state already conforms except for contrast, OR other issues are miraculously remedied by the same switch)? Or is there a rationale besides "conforming alternate version" that would cover this contrast switch fallback practice?

**Comment by GreggVan** [comment]
There needs to be a single page that meets WCAG.

if the “better contrast” page is a different page that meets only that SC — and not others — it is not a “conforming alternate version”

if the page conforms except for contrast and a button changes the contrast on that page (without breaking other SC in the process) (and that button is accessible itself) then that would pass. 

The reason that one page must pass all the SC is that people often have multiple disabilities. 

Does that answer the question?

**Comment by detlevhfischer** [comment]
@GreggVan 
> Does that answer the question?

 Yes, it does - but I would appreciate wider feedback indicating whether other auditors take the same strict line **or** would go for a PASS of 1.4.3 / 1.4.11 on pages that provide sufficient contrast via a switch while other FAILS exist on that page (as is usually the case). My hunch is that the latter approach is fairly common, but I may be wrong.

**Comment by mraccess77** [comment]
We have technique G174 https://www.w3.org/WAI/WCAG21/Techniques/general/G174

**Comment by fstrr** [comment]
This issue is labelled as a discussion, so we’re moving this to Discussions. There doesn’t seem to be an update to make to the documentation, but if that changes, we can move it back to the issues list.
